src/SFML/Window/Window.cpp

```diff
@@ -212,11 +219,10 @@ void Window::display()
         m_context->display();
 
     // Limit the framerate if needed
-    if (m_frameTimeLimit != Time::Zero)
-    {
-        sleep(m_frameTimeLimit - m_clock.getElapsedTime());
-        m_clock.restart();
-    }
+    sleep(m_frameTimeLimit - m_clock.getElapsedTime());
```

Comment:
`sf::sleep` does nothing if the requested duration is less than or equal to zero so we don't need an extra conditional here. That's already taken care of for us.

https://github.com/SFML/SFML/blob/9855552f6431016f1953e36b2f058fbd239ed725/src/SFML/System/Sleep.cpp#L46

Reply:
The first display call will still have a non-zero frame time since the timer starts upon construction of the window.

Reply:
Nevermind, you are right. The clock having microsecond precision, it seems very unlikely for elapsed time to be zero between window creation and first display call.

Reply:
In either case, I think `sleep` should be modified to instead read ``if (duration > Time::Zero)` because if the requested duration is zero, we don't want to be calling into all the actual sleeping logic.

Reply:
That's tricky. I think I remember `sleep(0)` still having the effect of signaling the OS's scheduler "I'm done for now". Less important these days with multiple cores, but just doing nothing in general might be the wrong approach here as it might impact perceived/expected behavior. Or in other words: I'm not sure `sleep(0)` should be a generic no-op.

Reply:
I put the conditional back out of an abundance of caution.